lib/service/servicecfg/config.go

```diff
@@ -286,6 +290,28 @@ type Config struct {
 	authServers []utils.NetAddr
 }
 
+// Option applies an option value for a Config.
+type Option interface {
+	Apply(any) error
+}
+
+// KubeMultiplexerConfigOption allows to modify multiplexer config for Kube Proxy TLS server's listener
+type KubeMultiplexerConfigOption func(any) error
+
+func (f KubeMultiplexerConfigOption) Apply(input any) error {
+	return f(input)
+}
+
+func NewKubeMultiplexerConfigOption(val bool) KubeMultiplexerConfigOption {
```

Comment:
nit: option names should be chosen s.t. the option's effect is obvious when looking at it in use.  `NewKubeMultiplexerConfigOption(true)` could mean anything.  How about something like `KubeMultiplexerShouldIgnoreSelfConnections(true)`?